_posts/deployment/2017-08-04-deploy-from-archive.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Deploy Directly from an Archive
+modified_at: 2017-08-04 00:00:00
+category: deployment
+tags: source deploy
+order: 1
+---
+
+A way to deploy your application on Scalingo is to use an url where your application code is upload on the format archive.
+
+## The Workflow
+
+If you want to deploy an application without git or github, you can put your application code in an archive `tar.gz`.
+
+Then you need to upload your application on a source, a repo on the web where your application will be conserved. If you do not know how do that, see the section [Create a Repo for the Archive]({% post_url deployment/2017-08-04-deploy-from-archive %}create-a-repo-for-the-archive).
+
+From your source, your repo, you must have a download_url that Scalingo can use to deploy your application.
+
+Thus, you have to provide your download_url on the Scalingo Api (https://api.scalingo.com/v1/apps/[:app]/deployments) with a request POST, see the section [Deployment]({% post_url deployment/2017-08-04-deploy-from-archive %}#to-deploy-an-archive).
+
+Once our platform receives your achive, the deployment begins.
+
+## To deploy an archive
+
+To deploy your archive, you have to make a POST request on `https://api.scalingo.com/v1/apps/[:app]/deployments`, like this:
+
+```bash
+curl -H "Accept: application/json" -H "Content-Type: application/json" -u ":$AUTH_TOKEN" \
+    -X POST https://api.scalingo.com/v1/apps/[example-app]/deployments -d \
+        '{
+            "deployment": {
+                "git_ref": "v0.0.2",
+                "download_url": "https://github.com/Scalingo/sample-go-martini/archive/master.tar.gz"
+            }
+        }'
+```
+
+* The `auth-token` is specific for Scalingo, you can find it on your [dashboard](https://my.scalingo.com/profile) or from our API. Every information are [here](https://developers.scalingo.com/index.html#authentication).
+
+* You have to replace `[example-app]` by your application name which must be created before. You can do that on our platform [my.scalingo.com](https://my.scalingo.com) or [from your shell](https://developers.scalingo.com/apps.html#create-an-application).
```

Comment:
~before~ in the first place
The link for "from your shell" is not exactly right. We can create an application from the shell using the CLI which is way easier than using some `curl` commands as pointed out by your link.